File: Arrays/209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        i = 0
        j = 0
        n = len(nums)
        sum = 0
        minlen = float(inf)

        while i < n and j < n:
            if sum < target:
                sum += nums[i]
                i += 1
            else:
                minlen = min(minlen, i - j)
                sum -= nums[j]
                j += 1
        
        while sum >= target :
            minlen = min(minlen, i - j)
            sum -= nums[j]
            j += 1

        return minlen if minlen != float(inf) else 0
    

File: Sliding_Window/209-minimum-size-subarray-sum/minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int i = 0, j = 0, n = nums.size(), sum = 0, minlen = INT_MAX;

        while(i < n && j < n){
            if(sum < target){
                sum += nums[i];
                i++;
            }
            else{
                minlen = min(minlen, i - j);
                sum -= nums[j];
                j++;
            }
        }
        while(sum >= target) {
            minlen = min(minlen, i - j);
            sum -= nums[j++];
        }

        while(sum >= target) {
            minlen = min(minlen, i - j);
            sum -= nums[j++];
        }

        return minlen == INT_MAX ? 0 : minlen;
    }
};